<a href="https://colab.research.google.com/github/Jay06eng/Machine-Learning/blob/main/Lab_8_Jay.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Part 1: CONCEPTUAL QUESTIONS

1. What differs data science from data analytics?

Data science tells why something happened and what will happen next using programming languages and data asnaslytics tells you the story of what happened. It "reads" the data and describes it.

2. Between mean and median, what do you think is more sensitive to outliers? Why?

The mean is more sensitive to ouliers as its calculated based on the value of each element, therefore when there are ouliers (too low or too high values) it directly affects the final result of the mean calculation. The median is just the number that divides the whole data set in the middle.

3. Is correlation also causation? Why or why not?

correlation means that two things move together while causation  is when one thing makes the other happen.

#Part 2: Correlation Coefficients

Interpret the following in terms of mean, std, and correlation coefficient:

In [ ]:
import numpy as np

# Manually create a dataset with 15 observations
coffee_consumption = [2, 4, 1, 5, 3, 1, 3, 2, 4, 5, 1, 4, 3, 2, 5]
productivity_scores = [60, 75, 50, 80, 70, 55, 65, 60, 72, 85, 48, 78, 68, 58, 90]

In [ ]:
print("the mean for  coffee consumption is ", np.mean(coffee_consumption))
print("the mean for  coffee consumption is ", np.mean(productivity_scores))
print("the standard deviation for coffee consumption is " , np.std(coffee_consumption))
print("the standard deviation for productivity scores is " , np.std(productivity_scores))
np.corrcoef(coffee_consumption, productivity_scores)


the mean for  coffee consumption is  3.0
the mean for  coffee consumption is  67.6
the standard deviation for coffee consumption is  1.4142135623730951
the standard deviation for productivity scores is  12.147976511885973


array([[1.        , 0.97401024],
       [0.97401024, 1.        ]])

The correlation is 0.97, which is very close to 1.

This means as coffee consumption goes up, productivity almost always goes up too in this dataset but it doesnt prove coffee causes.

In [ ]:
import numpy as np

# Manually create a dataset with 15 observations
hours_of_tv = [2, 3, 2, 4, 5, 1, 3, 2, 4, 5, 1, 4, 3, 2, 5]
grades = [78, 80, 75, 82, 85, 70, 79, 77, 81, 88, 69, 84, 79, 76, 87]


In [ ]:
dataset = np.column_stack((hours_of_tv, grades))
print(dataset)

[[ 2 78]
 [ 3 80]
 [ 2 75]
 [ 4 82]
 [ 5 85]
 [ 1 70]
 [ 3 79]
 [ 2 77]
 [ 4 81]
 [ 5 88]
 [ 1 69]
 [ 4 84]
 [ 3 79]
 [ 2 76]
 [ 5 87]]


In [ ]:
np.corrcoef(hours_of_tv, grades)

array([[1.        , 0.96404382],
       [0.96404382, 1.        ]])

The correlation is 0.96, which is very close to 1, meaning strong and positive correlation.

This means students who watch more TV in this dataset also tend to have higher grades.

But this is only a pattern in the numbers. It doesn’t mean TV causes better grades.



In [ ]:
import numpy as np

# Manually create a dataset with 15 observations
x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
y = [2, 4, 6, 8, 45, 12, 14, 45, 18, 20, 22, 24, 46, 28, 45]


In [ ]:
dataset_2 = np.column_stack((x, y))
print(dataset_2)

[[ 1  2]
 [ 2  4]
 [ 3  6]
 [ 4  8]
 [ 5 45]
 [ 6 12]
 [ 7 14]
 [ 8 45]
 [ 9 18]
 [10 20]
 [11 22]
 [12 24]
 [13 46]
 [14 28]
 [15 45]]


In [ ]:
np.corrcoef(x, y)

array([[1.        , 0.66128148],
       [0.66128148, 1.        ]])

The correlation is 0.66, which is a moderate positive relationship.

This means as x increases, y usually increases too, but not as tightly as the previous datasets.

The numbers move together, but there's more wiggle and less predictability.

#Part 3: Probability

1. Find the probability of getting a numbered card when a card is drawn from the pack of 52 cards.

“Numbered cards”  means 2,3,4,5,6,7,8,9,10.
That is 9 numbers in each suit.

4 suits * 9 numbered cards each = 36 numbered cards

Total cards = 52

𝑃(numbered card) = 36/52 = 9/13

The probability to draw a numbered card is 0.6923 or 69.23%

2. You have a bag containing 8 red balls, 5 green balls, and 3 blue balls. If you randomly draw two balls without replacement, what is the probability that both balls are green?

Red = 8
Green = 5
Blue = 3
Total = 8 + 5 + 3 = 16 balls.

We want green first and green second, without putting the first one back.

First ball green P(green1)= 5/16
1 green ball is gone after the firts draw, green balls left= 4, total balls left= 15
Prob is second ball green=4/15

Since both must happen we multiple=
P(both_green)=5/16*4/15=20/240=1/12

The probability to draw both balls green is 0.0833 or 8.3%


3. You roll a six-sided die. What is the probability of rolling an even number or a number greater than 4?

Die faces. 1,2,3,4,5,6.
Even numbers. 2,4,6
Numbers greater than 4. 5,6

All numbers that are even or greater than 4.
===>is. 2,4,5,6

That is 4 “good” outcomes out of 6 total.

P(4>)= 4/6=2/3 or 0.6666 or 66%
